### PR TITLE
Reorganize Table Of Content (#284)

### LINKED ISSUE #284: Table of contents is sub-optimal
It seems our documentation has grown substantially over the years and led to an overcrowded table of contents. While all the information is available, it might not be presented in the best possible way from a user perspective. Indeed, the table of contents currently include **34 items** which seems too many for a user to quickly find the desired information.
We also have a few instances of duplication of content that could be improved as it can be confusing for the reader.

It is proposed to simplify the current table of content by reorganizing it. It would mostly consist in consolidating similar items into a single main group. For instance, *User interfaces*, *Disks*, *Pools*, *Shares*, etc... could be grouped into a single header in the table of content; their individual page(s) structure would still remain the same, though.

Without a detailed analysis, it appears that reducing the number of first level headers in our table of contents to < 20 would already be a very big improvement. This number is arbitrary, though, as the final number should be decided by the logic that we want to follow for the table of content's organization.

The duplicated content could also be further simplified, although it could be part of a separate PR.

@phillxnet, before we go further into the details of what the new table of content organization could/should be, do you think it is an idea worth pursuing?

### PR DESCRIPTION
**Pull request by FroggyFlox**
Fixes #284.
@phillxnet, ready for review.

### This pull request's proposal
As discussed in #284, this pull-request (PR) reorganizes the docs in an effort to cut down on the number of top level headings and thus simplify how the information is presented to the user. The overall aim was of course to drastically improve clarity and ease to find information of interest.

The second biggest change in this PR is the addition of the sphinx extension [rediraffe](https://github.com/wpilibsuite/sphinxext-rediraffe) to automatically track and generate html redirects for all pages that were removed/renamed/deleted. All these redirects can be found in the `redirects.txt` file. @phillxnet, note that we will need to integrate rediraffe into our back-end so that the extension writes the redirects when running `make html`.

Another extension change is the deprecation of `sphinxcontrib-fulltoc` as this was not required anymore. The `layout.html` was adapted accordingly so that we can still show the full toc and not just the local toc.

@phillxnet, my apologies for not being able to break all these changes into multiple commits... I realize it makes it difficult to review.

### Summary of changes
- Deprecate the sphinxcontrib-fulltoc extension
- Add sphinxext-rediraffe extension (#284)
- Reorganize TOC to follow Web-UI structure
- Follow Docutils guideline for max line length of 79 char
- Write redirects with rediraffewritediff
- Remove unused Indices and Tables section in index.rst
- Delete ununsed image files

### Checklist
- [x] With the proposed changes no Sphinx errors or warnings are generated.
- [x] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

**Comment by FroggyFlox** [comment]
@phillxnet, sorry for the additional commits there; I should have tested that branch in a PR against my own fork to troubleshoot and fix our Github workflow there. Everything is now OK on this front and I'm glad I did update that as it highlighted an incorrect redirect for `update_channels.rst` (backslash instead of forward slash)--I'm liking the `rediraffecheckdiff` now :-) .

Everything should now be OK, though.

**Comment by phillxnet** [comment]
@FroggyFlox Just now having a look at this rather magnificent re-work re overall arrangement & build enhancement of our entire docs. Nice.

Looking good so far. Just noting some warnings re our backend patching system used to test this pr:
```
./290.patch:80565: trailing whitespace.
                    {%- endif %}                
./290.patch:80586: trailing whitespace.
  
./290.patch:80627: trailing whitespace.
  
./290.patch:80633: trailing whitespace.
  
./290.patch:80635: trailing whitespace.
  
warning: squelched 176 whitespace errors
warning: 181 lines add whitespace errors.
```
from the patch file presented by GitHub for this pr, retrieved via:
```
curl -L https://github.com/rockstor/rockstor-doc/pull/290.patch -o 290.patch
```
Don't make any changes on this pr based on this report as it's not worth it. We can always address these, if they are 'real', in thier own far less imoprtant issue.

I'll stagger on for now with our backend changes re:

> @phillxnet, note that we will need to integrate rediraffe into our back-end so that the extension writes the redirects when running `make html`.

Nearly there.

**Comment by phillxnet** [comment]
Getting one of our exiting important Web-UI embedded links from our local test webserver post pr merge:
```
wget http://build.lan/docs/update-channels/update_channels.html
...
HTTP request sent, awaiting response... 200 OK
Length: 521 [text/html]
Saving to: ‘update_channels.html’

update_channels.html                      100%[=====================================================================================>]     521  --.-KB/s    in 0s      

2021-07-26 19:00:09 (57.3 MB/s) - ‘update_channels.html’ saved [521/521]
```
Which is a browser based (read client side) redirect:

```
<html>
    <head>
        <noscript>
            <meta http-equiv="refresh" content="0; url=../interface/system/update_channels.html"/>
        </noscript>
    </head>
    <body>
        <script>
            window.location.href = '../interface/system/update_channels.html' + (window.location.search || '') + (window.location.hash || '');
        </script>
        <p>You should have been redirected.</p>
        <a href="../interface/system/update_channels.html">If not, click here to continue.</a>
    </body>
</html>

```

And when pasting the same link into a browser we are redirected to:

`
http://build.lan/docs/interface/system/update_channels.html
`

and we have server logs to reflect this successful forward:
```
 - - [26/Jul/2021:19:18:13 +0100] "GET /docs/update-channels/update_channels.html HTTP/1.1" 200 521 "-" ...
 - - [26/Jul/2021:19:18:14 +0100] "GET /docs/interface/system/update_channels.html HTTP/1.1" 200 19371 "http://build.lan/docs/update-channels/update_channels.html" ...
```

I see more now how rediraffe is doing this.

File: installation/quickstart.rst [hunk 1]
```diff
@@ -35,13 +40,6 @@ Rockstor `download options <https://rockstor.com/download.html>`_. Create a
 bootable CD or USB drive from the downloaded iso and proceed to the
 :ref:`installation` section.
 
-.. raw:: html
-
-   <div class="alert alert-info">
-   For convenience, you can also purchase a USB install disk from <a href="https://shop.rockstor.com/collections/diy-accessories/products/rockstor-usb-install-disk" target="_blank">our shop</a>.
-   Thanks for your support!
-   </div>
-
```
Comment (phillxnet) [inline installation/quickstart.rst:44]: Well that was a surprise. I was sure we had removed all of these links. Nice find / fix.

File: interface/docker-based-rock-ons/netdata_official.rst [hunk 2]
```diff
@@ -49,14 +50,14 @@ First, navigate to *System* > *Groups*, and search for the **docker** group in
 the bottom table (*Other system groups*). In the example below, the **docker**
 group has a :code:`GID` of :code:`476`.
 
-.. image:: ../images/docker-based-rock-ons/netdata_docker_group.png
+.. image:: /images/interface/docker-based-rock-ons/netdata_docker_group.png
```
Comment (phillxnet) [inline interface/docker-based-rock-ons/netdata_official.rst:53]: Thanks for standardising how we reference the "/images" directory. I'd not noticed the disparate use.

**Review by phillxnet — CHANGES_REQUESTED** [review]
@FroggyFlox Well that was an eye full. This is all fantastic stuff and our docs are in a much better state/format post these changes. I've only noticed only a single regression regarding our favouring internal references rather than external url's when linking to other part of the docs. I've marked it as a request change as I'm guessign this was an unintentional missed merge / rebase of sorts given the extensive amount of work involved in this pull request.

Let me know if you think we should skip this trivial regression and I'll merge as is and we can pick that up later.

Thanks for a massive amount of effort on this far reaching pull request and my apologies for not catching everything. It's rather impractical to review every change with so many file moves so I think we are best going with what little I've found and merging so we can take it in much smaller steps from there on :) .

File: howtos/home_nas_guide.rst [hunk 3]
```diff
@@ -25,57 +26,74 @@ Hardware
...
-      
+
 **A Server**
-          
-A server with at least the :ref:`minsysreqs`.
-          
-HP Proliant Microserver (Chosen for its small form factor, 4 easily accessible hard drive slots, and eSata capability to expand storage)
+
+A server with at least `minimum system requirements
+<https://rockstor.com/docs/quickstart.html#minimum-system-requirements>`_
```
[the remark this answers is not in the document]
Comment (FroggyFlox) [reply howtos/home_nas_guide.rst:33]: You are entirely correct... this must be the result of an incorrect rebase. I'll commit the changes manually to get these things in order.

[the remark this answers is not in the document]
Comment (FroggyFlox) [reply howtos/home_nas_guide.rst]: @phillxnet, fixed as of https://github.com/rockstor/rockstor-doc/pull/290/commits/5b48a2bc78884dc1d3797154bcc44d0bc5bcedd2

**Comment by FroggyFlox** [comment]
Thanks a lot for your review and catching that regression... I'm not sure how but it does look like a screw-up while rebasing, indeed... Whatever happened for the internal link, I switched it back to what it should be so it should be good this time.

Thanks again!

**Review by phillxnet — APPROVED** [review]

**Comment by phillxnet** [comment]
@FroggyFlox This is quite the improvement. Thanks for all your efforts on this one. I'm bound to have missed some issues in it's review and apologies for that in advance but it's clearly, for me, a welcome step-up in our docs overall. They seemed to have gotten into quite the disarray unfortunately. They now look far less intimidating and 'wall-of-text' like, and more like a hand-book which is perfect.

Chuffed to merge and will then experimentally publish and test the live https variant. Should all be good thought from my local http tests.

For those concerned about our adoption here of client side redirects, it is appreciated that this is sub-optimal; however in the light of maintaining so many, potentially unending, redirects over time the value of preserving links baked into our older Web-UI instances, and forum numerous  threads, out weights the relatively cheap 'cost' of a repo (git) maintained redirection system.

I would like to join @FroggyFlox in his praise / appreciation of the fine work done by the [rediraffe](https://github.com/wpilibsuite/sphinxext-rediraffe) team who helped us make this major reorganisation of our docs possible.

**Comment by phillxnet** [comment]
@FroggyFlox The result of this pull request's merge is now published to production. Looks nice there.

**Pull request merged**